src/main/java/org/broadinstitute/dsm/db/dto/ddp/participant/ParticipantDataDto.java

```diff
@@ -1,29 +1,91 @@
 package org.broadinstitute.dsm.db.dto.ddp.participant;
 
+import java.io.IOException;
 import java.lang.reflect.Type;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 
+import com.fasterxml.jackson.annotation.*;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
 import com.google.gson.reflect.TypeToken;
 import lombok.Setter;
 import org.apache.commons.lang3.StringUtils;
-
+import org.broadinstitute.dsm.db.dao.ddp.participant.ParticipantDataDao;
+import org.broadinstitute.dsm.db.structure.ColumnName;
+import org.broadinstitute.dsm.db.structure.TableName;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.util.ObjectMapperSingleton;
+
+@TableName(
+        name = DBConstants.DDP_PARTICIPANT_DATA,
+        alias = DBConstants.DDP_PARTICIPANT_DATA_ALIAS,
+        primaryKey = DBConstants.PARTICIPANT_DATA_ID,
+        columnPrefix = "")
+@JsonIgnoreProperties(ignoreUnknown = true)
 @Setter
 public class ParticipantDataDto {
 
     private static final Gson gson = new Gson();
 
+    @ColumnName(ParticipantDataDao.PARTICIPANT_DATA_ID)
     private int participantDataId;
+
+    @ColumnName(ParticipantDataDao.DDP_PARTICIPANT_ID)
     private String ddpParticipantId;
+
+    /*
+        used only for Jackson library, jackson by default uses getter of the field to serialize its data
+        since we follow Optional way of getters, jackson by default weirdly serializes Optional and not field's data
+        JsonGetter is used to give a hint to Jackson to use specific method to serialize data for the field
+     */
+    @JsonGetter("ddpParticipantId")
+    private String serializeDdpParticipantId() {
+        return ddpParticipantId;
+    }
+
+    @ColumnName(ParticipantDataDao.DDP_INSTANCE_ID)
     private int ddpInstanceId;
+
+    @ColumnName(ParticipantDataDao.FIELD_TYPE_ID)
     private String fieldTypeId;
+
+    /*
+        used only for Jackson library, jackson by default uses getter of the field to serialize its data
+        since we follow Optional way of getters, jackson by default weirdly serializes Optional and not field's data
+        JsonGetter is used to give a hint to Jackson to use specific method to serialize data for the field
+     */
+    @JsonGetter("fieldTypeId")
+    private String serializeFieldTypeId() {
+        return fieldTypeId;
+    }
+
+    @ColumnName(ParticipantDataDao.DATA)
+    @JsonProperty("dynamicFields")
     private String data;
+
+    @JsonProperty("dynamicFields")
```

Comment:
does it make a problem if there are two JsonProperties with the same name?

Reply:
No